src/test/util/mocks.cpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2019 The Unit-e developers
```

Comment:
I'd move all implementation to `mocks.h` and delete this file. I feel `mocks.h` is head-only library, and I don't see any strong reason to move a couple of small pieces of them in independent translation unit, just because we can.

Reply:
In fact, having all implementations in one place, simplifies reading.

Reply:
Not all of the implementations can be defined inline as there's a dependency between `Mock` and `MethodMockBase`. This requires the forward declaration of `MethodMockBase` and a separate implementation of the `Mock` methods. Since some already had to go somewhere (yeah they could be defined in the header file as well but separately) I didn't feel to bad about moving them. Especially since their implemenations do not really help a reader to understand anything better here (IMO), as it's outlined in comments and the method names are pretty descriptive.